### PR TITLE
Orangered fixups

### PR DESCRIPTION
**Pull request by andytuba**
mostly to ensure the native message count is up to date with the nre floating message, and unreadLinksToInbox is applied to every orangered element.

### CODE REVIEW

File: lib/modules/orangered.js [hunk 1]
```diff
@@ -171,6 +167,18 @@ const setupFaviconBadge = _.once(() => {
 	return favicon;
 });
 
+function setupNativeButtons() {
```
Comment (erikdesjardins) [inline lib/modules/orangered.js]: Maybe replace this and two of the globals with something like

```js
const nativeButtons = _.once(() => {
    const inbox = document.querySelector('#header-bottom-right #mail');
    let mailCount = document.querySelector('#header-bottom-right .message-count');
    if (!mailCount && (module.options.updateCurrentTab.value || module.options.updateOtherTabs.value)) {
        mailCount = document.createElement('a');
        mailCount.style = 'display: inline-none;';
        $(inbox).after(mailCount);
    }
    return { inbox, mailCount };
});
```

Also `inline-none`?

Comment (andytuba) [inline lib/modules/orangered.js]: i think you are right about the inline-none

Comment (andytuba) [inline lib/modules/orangered.js]: the once is also a good idea

File: lib/modules/orangered.js [hunk 2]
```diff
@@ -171,9 +166,32 @@ const setupFaviconBadge = _.once(() => {
...
+	...nativeButtons(),
+	...floatingButtons(),
+	$modmailButton: $('#modmail'),
+});
+
+const $orangeredElements = () => {
+	const dict = orangeredElements();
+	const elements = Object.keys(dict).map(key => dict[key]);
+	return elements.reduce(($all, ele) => $all.add(ele), $());
+};
```
Comment (andytuba) [inline lib/modules/orangered.js]: @erikdesjardins How's this read? Feels a little awkward, but it's really only called once.

File: lib/modules/orangered.js [hunk 3]
```diff
@@ -171,9 +166,32 @@ const setupFaviconBadge = _.once(() => {
...
-	if ($floatingInboxCount) return;
+const orangeredElements = () => ({
+	...nativeButtons(),
+	...floatingButtons(),
+	$modmailButton: $('#modmail'),
+});
+
+const $orangeredElements = () => {
+	const dict = orangeredElements();
+	const elements = Object.keys(dict).map(key => dict[key]);
```
Comment (erikdesjardins) [inline lib/modules/orangered.js]: This is `Object.values(dict)`, I think :stuck_out_tongue_winking_eye:

File: lib/modules/orangered.js (hunk 2 above)
Comment (erikdesjardins) [inline lib/modules/orangered.js]: Yeah that looks fine

File: lib/modules/orangered.js [hunk 4]
```diff
@@ -171,9 +166,32 @@ const setupFaviconBadge = _.once(() => {
...
+	return elements.reduce(($all, ele) => $all.add(ele), $());
+};
+
+const nativeButtons = _.once(() => {
+	const nativeInboxButton = document.querySelector('#header-bottom-right #mail');
+	let nativeInboxCount = document.querySelector('#header-bottom-right .message-count');
+	if (!nativeInboxCount && (module.options.updateCurrentTab.value || module.options.updateOtherTabs.value)) {
+		nativeInboxCount	= document.createElement('a');
```
Comment (erikdesjardins) [inline lib/modules/orangered.js]:
```
nativeInboxCount = document
                ^ looks like this is a tab
```

File: lib/modules/orangered.js [hunk 5]
```diff
@@ -171,9 +166,32 @@ const setupFaviconBadge = _.once(() => {
 	return favicon;
 });
 
-function setupFloatingButtons() {
-	if (!module.options.showFloatingEnvelope.value) return;
-	if ($floatingInboxCount) return;
+const orangeredElements = () => ({
```
Comment (erikdesjardins) [inline lib/modules/orangered.js]: You could also wrap this in `_.once` if you want so you aren't selecting `#modmail` every time

File: lib/modules/orangered.js (hunk 3 above)
Comment (andytuba) [inline lib/modules/orangered.js]: Only in Firefox 😭

File: lib/modules/orangered.js (hunk 4 above)
Comment (andytuba) [inline lib/modules/orangered.js]: huh.

File: lib/modules/orangered.js (hunk 3 above)
Comment (erikdesjardins) [inline lib/modules/orangered.js]: Nah Babel transforms it https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/1ef2c155b88c2dbbdd4a36d2b97ce1947df239ba/lib/core/profiling.js#L99-L101

File: lib/modules/orangered.js (hunk 5 above)
Comment (andytuba) [inline lib/modules/orangered.js]: I'll sidestep that: https://github.com/honestbleeps/Reddit-Enhancement-Suite/pull/3299/commits/42d273a278c85e573537a82646ec1e5cad5b5f30

File: lib/modules/orangered.js (hunk 3 above)
Comment (andytuba) [inline lib/modules/orangered.js]: Neato! https://github.com/honestbleeps/Reddit-Enhancement-Suite/pull/3299/commits/cacaebc0bccf66721825e2bae8d8ae02c4cc78f0

**Pull request merged**